app/packages/core/src/components/Actions/Similar.tsx

```diff
@@ -237,91 +291,201 @@ const SortBySimilarity = React.memo(
+                    size="small"
+                    disableRipple
+                    sx={{ color: "#0288d1" }}
+                  >
+                    <InfoIcon />
+                  </IconButton>
+                </div>
+              </Tooltip>
+              <Tooltip text="Advanced settings" placement={"top-center"}>
+                <div ref={refSetting} onClick={() => setOpen((o) => !o)}>
+                  <IconButton aria-label="settings" size="small">
+                    <SettingsIcon />
+                  </IconButton>
+                </div>
+              </Tooltip>
+            </ButtonGroup>
+          </div>
+        )}
+        {(!state.brainKey ||
+          (!isImageSearch && hasKeyButNotSupportTextPrompts)) && (
           <>
-            <Input
-              placeholder={"k (default = None)"}
-              validator={(value) => value === "" || /^[0-9\b]+$/.test(value)}
-              value={state.k === null ? "" : String(state.k)}
-              setter={(value) => {
-                setParameter("k", value === "" ? null : Number(value));
-              }}
-            />
-            <Input
-              placeholder={"dist_field (default = None)"}
-              validator={(value) => !value.startsWith("_")}
-              value={state.distField === null ? "" : state.distField}
-              setter={(value) => {
-                setParameter("distField", value === "" ? null : value);
-              }}
-            />
-            <Checkbox
-              name={"reverse"}
-              value={Boolean(state.reverse)}
-              setValue={(v) => setParameter("reverse", v)}
-            />
-            <PopoutSectionTitle style={{ fontSize: 14 }}>
-              Brain key
+            {hasKeyButNotSupportTextPrompts && (
+              <PopoutSectionTitle style={{ fontSize: 12 }}>
+                No brain keys support text prompts
+              </PopoutSectionTitle>
+            )}
+            <PopoutSectionTitle>
+              <ActionOption
+                href={SORT_BY_SIMILARITY}
+                text={"Search by text similarity"}
+                title={"About sorting by similarity"}
+                style={{
+                  background: "unset",
+                  color: theme.text.primary,
+                  paddingTop: 0,
+                  paddingBottom: 0,
+                }}
+                svgStyles={{ height: "1rem", marginTop: 7.5 }}
+              />
             </PopoutSectionTitle>
-            <RadioGroup
-              choices={choices.choices}
-              value={state.brainKey}
-              setValue={(v) => setParameter("brainKey", v)}
-            />
-            {state.brainKey && (
-              <>
-                <PopoutSectionTitle></PopoutSectionTitle>
+          </>
+        )}
+        {open && hasValidKeys && (
+          <div>
+            <div>
+              Find the{" "}
+              <div style={{ width: "30px", display: "inline-block" }}>
```

Comment:
Please try using flex positioning

Reply:
Here, 30px is exactly the size we want. And we don't want the input to stretch. The size of the Popout is fixed, too. 
![Screenshot 2023-02-10 at 12 12 07 PM](https://user-images.githubusercontent.com/17770824/218165892-826676a0-a8c7-448f-9c39-f38008e6e3b7.png)